dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClusterCPUChartTask.java

```diff
@@ -0,0 +1,157 @@
+ * on <a
+ * href="https://docs.cloudera.com/documentation/enterprise/latest/topics/cm_dg_tsquery.html">tsquery</a>
+ * language.
+ */
+public class ClouderaClusterCPUChartTask extends AbstractClouderaManagerTask {
+  enum TimeSeriesAggregation {
+    RAW,
+    TEN_MINUTELY,
+    HOURLY,
+    SIX_HOURLY,
+    DAILY,
+    WEEKLY,
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaCMFHostsTask.class);
+  /*
+    SELECT cpu_percent_across_hosts WHERE entityName = "1546336862" AND category = CLUSTER
+  */
+  private static final String TS_CPU_QUERY_TEMPLATE =
+      "SELECT cpu_percent_across_hosts WHERE entityName = \"%s\" AND category = CLUSTER";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final ZonedDateTime from;
+  private final TimeSeriesAggregation tsAggregation;
+
+  public ClouderaClusterCPUChartTask() {
+    this(Instant.now().atZone(ZoneId.of("UTC")), TimeSeriesAggregation.RAW);
+  }
+
+  public ClouderaClusterCPUChartTask(ZonedDateTime from, TimeSeriesAggregation tsAggregation) {
+    super(
+        String.format(
+            "cmf-cluster-cpu-%s-%s.jsonl",
+            from.toString().replaceAll("[-:\\[\\]ZUTC.]", "").toLowerCase(),
+            tsAggregation.toString().toLowerCase()));
+    this.from = from;
+    this.tsAggregation = tsAggregation;
+  }
+
+  @Override
+  protected Void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    List<ClouderaClusterDTO> clusters = getClustersFromHandle(handle);
+
+    final String timeSeriesAPIUrl = getTimeSeriesUrl(handle.getApiURI().toString());
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      for (ClouderaClusterDTO cluster : clusters) {
+        String cpuPerClusterQuery = getQueryToFetchCPUTimeSeriesOnCluster(cluster.getId());
+        LOG.debug(
+            "Execute charts query: [{}] for the cluster: [{}].",
+            cpuPerClusterQuery,
+            cluster.getName());
+
+        LOG.debug(this.tsAggregation.toString());
+
+        URIBuilder uriBuilder = new URIBuilder(timeSeriesAPIUrl);
+        uriBuilder.addParameter("query", cpuPerClusterQuery);
+        uriBuilder.addParameter("desiredRollup", this.tsAggregation.toString());
+        uriBuilder.addParameter("mustUseDesiredRollup", "true");
+        uriBuilder.addParameter("from", getISODateTime(this.from));
+
+        try (CloseableHttpResponse chart = httpClient.execute(new HttpGet(uriBuilder.build()))) {
+          JsonNode jsonNode = objectMapper.readTree(chart.getEntity().getContent());
+          writer.write(jsonNode.toString());
+          writer.write('\n');
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private List<ClouderaClusterDTO> getClustersFromHandle(@Nonnull ClouderaManagerHandle handle) {
```

Comment:
`get` used for some straight forward logic or getting the same instance of object. In this case `build` or `create` will be better, because there is some logic in the method and if this method will be call several time, different instances will be returned.